mava/networks/torsos.py

```diff
@@ -35,11 +36,16 @@ def setup(self) -> None:
     def __call__(self, observation: chex.Array) -> chex.Array:
         """Forward pass."""
         x = observation
-        for layer_size in self.layer_sizes:
+        for i, layer_size in enumerate(self.layer_sizes):
             x = nn.Dense(layer_size, kernel_init=orthogonal(np.sqrt(2)))(x)
             if self.use_layer_norm:
                 x = nn.LayerNorm(use_scale=False)(x)
-            x = self.activation_fn(x)
+
+            if i != len(self.layer_sizes) - 1:
+                x = self.activation_fn(x)
+            elif i == len(self.layer_sizes) - 1 and self.activate_final:
+                x = self.activation_fn(x)
```

Comment:
Little cleaner I think
```suggestion
            should_activate = (i < len(self.layer_sizes) - 1) or self.activate_final
            x = self.activation_fn(x) if should_activate else x
```